File: @LV1/1.3/sevenwonders/sevenwonders.hs
count :: String -> Char -> Int
count s c = length $ filter (== c) s

calcCard :: [Int] -> Int
calcCard cnt = sum [ x * x | x <- cnt ]

main :: IO ()
main = do
  s <- getLine
  let deck = "TCG"
      cnt  = map (count s) deck
  print $ calcCard cnt + 7 * minimum cnt

File: @LV1/1.3/sevenwonders/sevenwonders.rb
DECK = ['T', 'C', 'G']

find_min = ->(s) { DECK.map(&->(x) { s.count(x) } ).min }
calc_card = ->(s) { DECK.map(&-> (x) { s.count(x)**2 }).sum }

card = gets
puts calc_card.call(card) + 7*find_min.call(card)
